dll/shellext/zipfldr/zipfldr.cpp

```diff
@@ -174,12 +174,17 @@ STDAPI DllUnregisterServer()
 
 EXTERN_C
 BOOL WINAPI
-RouteTheCall(IN HWND hWndOwner, IN HINSTANCE hInstance, IN PCWSTR lpStringArg, IN INT Show)
+RouteTheCallW(IN HWND hWndOwner, IN HINSTANCE hInstance, IN PCWSTR lpStringArg, IN INT Show)
 {
     CStringW path = lpStringArg;
     PathRemoveBlanksW(path.GetBuffer());
     path.ReleaseBuffer();
     path = L"\"" + path + L"\"";
-    ShellExecuteW(NULL, L"open", L"explorer.exe", path.GetString(), NULL, SW_SHOWNORMAL);
+
+    WCHAR app[MAX_PATH];
+    GetWindowsDirectoryW(app, _countof(app));
+    PathAppendW(app, L"explorer.exe");
+
+    ShellExecuteW(NULL, L"open", app, path.GetString(), NULL, Show ? Show : SW_SHOWNORMAL);
```

Comment:
Just using whatever is given to "Show" is not enough?

Reply:
If some app does ShellExecute with all 0's except the file parameter then it would tell Explorer to open a hidden window. It is rather unlikely I admit.

Reply:
This can be verified by an API test, right?

Reply:
I don't know what are you really going to test. It is [well known](https://devblogs.microsoft.com/oldnewthing/20100301-00/?p=14773) that the show command used to start a new process will force the show mode on the first call to `ShowWindow` (in certain scenarios).

If you do `ShellExecuteW(0, 0, L"notepad.exe", 0, 0, 0);` then you will start a new Notepad process that is hidden. I could not reproduce the hidden window on a .zip in my simple test but I think this is because ROS Explorer does not support the separate process option yet and therefore the show command is lost when it reuses the main process to open the new window.

I'd rather just remove the check rather than try to come up with some kind of test...

Reply:
I've created a simple API test (screenshot below), and it seems on Windows it completely ignores the Show parameter. The new explorer window is always visible and maximized.

![image](https://github.com/reactos/reactos/assets/578406/b139b9b1-1ee5-4ed9-ad31-0b41122fa876)

Reply:
Alright, let's keep this as it is for now.